docs/contributing_modules.md

```diff
@@ -0,0 +1,129 @@
+# Contributing Modules
+
+BenchPress is designed to be extensible. You can create your own modules and contribute them to the community.
+
+## Module Structure
+
+A valid module must have these files;
+
+```text
+Modules/BenchPress.Azure/Public
+└───Confirm-{Resource Type Name}.ps1
+
+Modules/BenchPress.Azure/Tests/Public
+└───Confirm-{Resource Type Name}.Tests.ps1
+
+examples/{Resource Type Name}
+├───{Resource Type Name}.Tests.ps1
+├───{Resource Type Name}.bicep
+└───{Resource Type Name}_Example.md
+```
+
+## Module Development
+
+- Create a new module with the required files mentioned above.
+
+- Add `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in the alphabetical order.
+
+- Add `{Resource Type Name}` to the [Modules/BenchPress.Azure/Classes/ResourceType.psm1](../Modules/BenchPress.Azure/Classes/ResourceType.psm1) file in the alphabetical order.
+
+- Add `. $PSScriptRoot/Confirm-{Resource Type Name}.ps1` to the beginning of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in the alphabetical order.
+
+- Add a section for the new resource type to the end of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in the alphabetical order, such as;
+
+```powershell
+"{Resource Type Name}" {
+  return Confirm-{Resource Type Name} -ResourceGroupName $ResourceGroupName -Name $ResourceName
+}
+```
+
+## Module Testing
+
+- Run the following command in the root directory of the project to create local copy of the BenchPress in the `bin` folder;
+
+```powershell
+.\build.ps1 -Inline
+```
+
+- Replace the `BeforeAll` section of the `examples/{Resource Type Name}/{Resource Type Name}.Tests.ps1` file with the following code;
+
+```powershell
+# old
+Import-Module Az.InfrastructureTesting
+
+# new
+Import-Module ../../bin/BenchPress.Azure.psd1
+```
+
+- Run this command to and create a new service principal and store the credentials in a variable;
```

Comment:
```suggestion
- Run this command to create a new service principal and store the credentials in a variable;
```

Reply:
moved this section to the getting_started.md file, thanks 👍